modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/01-object-definition.batch-engine-data.json

```diff
@@ -15,6 +15,7 @@
 	},
 	"items": [
 		{
+			"defaultLanguageId": "en_US",
```

Comment:
Is this mandatory now? How can we ensure we do not forget to add it to new developments? Will it break at runtime? Should we have a SF rule for it?

Reply:
When creating an object definition, it is necessary to provide a label in the site default language. Since these _modifiable_ system object definitions only provide an English label, we are now using the `defaultLanguageId` (English) to populate the site default locale when it's missing.

So without this, it breaks when the default locale is not English.

About SF rules, I am not sure when they are applicable/adequate. We also have the json's of the _unmodifiable_ system object definitions (.e.g. Account, User etc.), which should not have the `defaultLanguageId` specified in the json.

Perhaps if the current examples have this field, we are likely to add it to new developments as well.

Reply:
@devpedrotavares I think it's OK to forward this. Just make sure in the final PR you mention @brianchandotcom and let him know that this should be SF enforced.

Reply:
All right, thank you! Should we move on from this repository or from Object's ?